### PR TITLE
[improve](explode-json-array)change explode-json-array-xx func signature from string to json type

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes
change explode-json-array-xx func signature from string to json type to improve function behavior 
Issue Number: close #xxx

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/generator/ExplodeJsonArrayDouble.java [hunk 1]
```diff
@@ -23,7 +23,7 @@
 import org.apache.doris.nereids.trees.expressions.shape.UnaryExpression;
 import org.apache.doris.nereids.trees.expressions.visitor.ExpressionVisitor;
 import org.apache.doris.nereids.types.DoubleType;
-import org.apache.doris.nereids.types.VarcharType;
+import org.apache.doris.nereids.types.JsonType;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/generator/ExplodeJsonArrayDouble.java:26]: Isn't it necessary to modify the code of the Backend (BE) as well?

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

File: be/src/vec/exprs/table_function/vexplode_json_array.cpp [hunk 2]
```diff
@@ -59,10 +61,12 @@ void VExplodeJsonArrayTableFunction<DataImpl>::process_row(size_t row_idx) {
 
     StringRef text = _text_column->get_data_at(row_idx);
     if (text.data != nullptr) {
-        rapidjson::Document document;
```
Comment (eldenmoon) [inline be/src/vec/exprs/table_function/vexplode_json_array.cpp:62]: will result compability issue？ eg. upgrade fe but be not upgrate be

Comment (amorynan) [reply be/src/vec/exprs/table_function/vexplode_json_array.cpp:62]: I see, we should keep string can also be kept

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

File: be/src/vec/exprs/table_function/vexplode_json_array.cpp [hunk 3]
```diff
@@ -59,17 +62,28 @@ void VExplodeJsonArrayTableFunction<DataImpl>::process_row(size_t row_idx) {
...
-        document.Parse(text.data, text.size);
-        if (!document.HasParseError() && document.IsArray() && document.GetArray().Size()) {
-            _cur_size = _parsed_data.set_output(document, document.GetArray().Size());
+        if (WhichDataType(_text_datatype).is_json()) {
```
Comment (morningman) [inline be/src/vec/exprs/table_function/vexplode_json_array.cpp:65]: Maybe we should use template?

**Review by eldenmoon — APPROVED** [review]
LGTM

**Comment by amorynan** [comment]
run cloud_p0

File: be/src/vec/exprs/table_function/vexplode_json_array.cpp (hunk 3 above)
Comment (amorynan) [reply be/src/vec/exprs/table_function/vexplode_json_array.cpp:65]: here using a same sign with old rapid , so can not use template, just according data

**Comment by hello-stephen** [comment]
run compile

**Comment by amorynan** [comment]
run buildall

**Review by morningman — APPROVED** [review]

**Comment by amorynan** [comment]
run p0

**Comment by xiaokang** [comment]
behavior change is fixed by #39572

**Pull request merged**